docs/source/developer_guide/core/caching.rst

```diff
@@ -3,10 +3,30 @@ Caching: implementation details
 
 This section covers some details of the caching mechanism which are not discussed in the :ref:`user guide <caching>`. If you are developing a plugin and want to modify the caching behavior of your classes, we recommend you read :ref:`this section <caching_matches>` first.
 
+Controlling caching
+-------------------
+
+When developing calculation and data classes, there are some methods you can use to determine how the hash is created:
+
+* To ignore specific attributes, a ``Node`` subclass can have a ``_hash_ignored_attributes`` attribute. This is a list of attribute names which are ignored when creating the hash.
+* For calculations, the ``_hash_ignored_inputs`` attribute lists inputs that should be ignored when creating the hash.
+* To add things which should be considered in the hash, you can override the :meth:`_get_objects_to_hash <aiida.orm.nodes.Node._get_objects_to_hash>` method. Note that doing so overrides the behavior described above, so you should make sure to use the ``super()`` method.
+* Pass a keyword argument to :meth:`.get_hash <aiida.orm.nodes.Node.get_hash>`. These are passed on to ``aiida.common.hashing.make_hash``. For example, the ``ignored_folder_content`` keyword is used by the :class:`JobCalculation <aiida.orm.nodes.process.calculation.calcjob.CalcJobNode>` to ignore the ``raw_input`` subfolder of its repository folder.
```

Comment:
encapsulate "aiida.common.hashin.make_hash" in a :func: as well, or similar.

Reply:
Class "JobCalculation" doesn't exist anymore. Just use the tilde method here, I think.

Reply:
Do we still have a subfolder called "raw_input"?

Reply:
I think the "raw_inputs" is removed and now input files are stored at the top level of a `CalcJobNode` #3125

Reply:
thanks for bringing this up. will fix this once #3125 is resolved.

Reply:
The `raw_input` sub folder is still there, but its existence is hidden from the user and the `Repository` interface automatically deals with it

Reply:
@sphuber So, do I understand correctly that the sentence "For example, ..." is still accurate (after fixing the class reference)?

I see this is still in the code
```python
    def get_hash(self, ignore_errors=True, ignored_folder_content=('raw_input',), **kwargs):  # pylint: disable=arguments-differ
```

Reply:
Ah, we must have missed this in PR #3130 . I guess this is how the problem of mutable `CalcJobNode` hash was dealt with. I suspect it was actually working but was broken by PR #3108 . That dealt with a similar problem, where the hash computed before storing, changed after storing, because the point of "cleaning" attributes was changed. To fix this, @giovannipizzi made the `get_hash` function not callable on unstored nodes, to prevent people from getting a hash, that would change once stored. The `store` method then calls the `_get_hash` method. The overridden `CalcJobNode.get_hash` however now never gets called by the `store` method and so the exclusion of the `raw_input` folder from the hash was circumvented. I think, since we solved the problem differently now, we can simply remove the `CalcJobNode.get_hash` override

Reply:
Opened #3312 for this

Reply:
This has been resolved

Reply:
Coming back to this: The sentence
> For example, the ``ignored_folder_content`` keyword is used by the :class:`JobCalculation <aiida.orm.nodes.process.calculation.calcjob.CalcJobNode>` to ignore the ``raw_input`` subfolder of its repository folder.

now no longer applies, and I will simply delete it.